### ISSUE TITLE
make gemini support 

### ISSUE THREAD

**Issue by spline2hg** [UTSAVS26/Personalized-News-Aggregator#1]
can we create support for gemini api,its easy to get started with gemini api key

**Comment by UTSAVS26** [comment]
@spline2hg 
Thanks for the suggestion! I’ll consider adding Gemini API support. If you'd like, feel free to raise a detailed issue and contribute by working on it as well.

**Comment by spline2hg** [comment]
yeah cool, actually i want to run ur project locally but i dont have credit card for open ai api key ,is there any other way

**Comment by amannaik247** [comment]
Hello, can I work on this issue and set up , gemini or groq api ? If you guys don't want to use gpt for testing

**Comment by UTSAVS26** [comment]
> yeah cool, actually i want to run ur project locally but i dont have credit card for open ai api key ,is there any other way

I have used openAPI key itself they provide free too as much I know

**Comment by UTSAVS26** [comment]
> Hello, can I work on this issue and set up , gemini or groq api ? If you guys don't want to use gpt for testing 

You can work with @spline2hg, if you want to.

**Comment by amannaik247** [comment]
open ai has limited free resources, unlike groq or gemini.

**Comment by UTSAVS26** [comment]
> open ai has limited free resources, unlike groq or gemini.

Okie then you can use that i don't have any concern regarding it.

**Comment by amannaik247** [comment]
can you tag this as hacktoberfest2024?

**Comment by amannaik247** [comment]
can you tag this as hacktoberfest2024?

**Comment by UTSAVS26** [comment]
Whom i should assign this issue to? @amannaik247 or @spline2hg

**Comment by spline2hg** [comment]
hey @UTSAVS26 i can work on creating support of gemini api .

**Comment by UTSAVS26** [comment]
> hey @UTSAVS26 i can work on creating support of gemini api .

Okie then I am assigning this usse to you

**Comment by UTSAVS26** [comment]
> > hey @UTSAVS26 i can work on creating support of gemini api .
> 

Okie then I am assigning this usse to you
Under which label? `hacktoberfest`

**Comment by spline2hg** [comment]
> > > hey @UTSAVS26 i can work on creating support of gemini api .
> 
> Okie then I am assigning this usse to you Under which label? `hacktoberfest`

yes hacktoberfest label seems ok
